src/sage/groups/generic.py

```
@@ -982,7 +986,8 @@ def discrete_log(a, base, ord=None, bounds=None, operation='*', identity=None, i
                     c = bsgs(gamma, h, (0, temp_bound), inverse=inverse, identity=identity, op=op, operation=operation)
                 elif algorithm == 'rho':
                     c = discrete_log_rho(h, gamma, ord=pi, inverse=inverse, identity=identity, op=op, operation=operation)
-                elif algorithm == 'lambda':
+                else:
+                    assert algorithm == 'lambda'
```

Same here: better to raise an error.